core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -137,7 +137,18 @@ object TestUtils extends Logging {
     val parentFile = new File(parent)
     parentFile.mkdirs()
 
-    JTestUtils.tempDirectory(parentFile.toPath, null)
+    val result = JTestUtils.tempDirectory(parentFile.toPath, null)
+
+    parentFile.deleteOnExit()
+    Exit.addShutdownHook("delete-temp-log-dir-on-exit", {
+      try {
+        Utils.delete(parentFile)
```

Comment:
Do we need both `parentFile.deleteOnExit()` and `Utils.delete(parentFile)` on the shutdown hook?

Reply:
Good point - turns we can just use `parentFile.deleteOnExit()`. Updated in [394f2e3](https://github.com/apache/kafka/pull/15289/commits/394f2e30e3ac2110921f8e1ea2978c405194a11b)